panic: invalid score report

**Describe the bug**

cnspec panics if two policies use the same query uid

**To Reproduce**

Create the following policy:

```
policies:
  - uid: example1
    name: Example policy 1
    version: "1.0.0"
    groups:
      - filters: asset.family.contains('unix')
        checks:
          - uid: sshd-01
  - uid: example2
    name: Example policy 2
    version: "1.0.0"
    groups:
      - filters: asset.family.contains('unix')
        checks:
          - uid: sshd-01



queries:
  - uid: sshd-01
    title: Ensure the port is set to 22
    mql: sshd.config.params["Port"] == 22
```

And try to run it:

```
cnspec scan local -f panic.mql.yaml                                                                                                                                                                                                                                                                                                                                                                                                         2m chris-rock/aix-policies[7ef46bd6]
→ loaded configuration from /Users/chris/.config/mondoo/mondoo.yml using source default
! No credentials provided. Switching to --incognito mode.
→ discover related assets for 1 asset(s)
panic: invalid score report

goroutine 72 [running]:
go.mondoo.com/cnspec/v9/policy/executor/internal.(*ReportingJobNodeData).consume(0x140002b1800, {0x14000823620, 0xc}, 0x14001239aa0)
        /Users/chris/go/src/go.mondoo.com/cnspec/policy/executor/internal/nodes.go:442 +0x1c4
go.mondoo.com/cnspec/v9/policy/executor/internal.(*GraphExecutor).Execute(0x14000e414f0)
        /Users/chris/go/src/go.mondoo.com/cnspec/policy/executor/internal/graph.go:137 +0x700
go.mondoo.com/cnspec/v9/policy/executor.ExecuteResolvedPolicy({0x101c0d4e0, 0x14000212b60}, {0x101c10050?, 0x14001b736b0}, {0x140007c8900, 0x3a}, 0x129b72060?, {0x140002a7838, 0x2, 0x8}, ...)
        /Users/chris/go/src/go.mondoo.com/cnspec/policy/executor/graph.go:47 +0x40c
go.mondoo.com/cnspec/v9/policy/scan.(*localAssetScanner).runPolicy(0x14000d35d38)
        /Users/chris/go/src/go.mondoo.com/cnspec/policy/scan/local_scanner.go:1000 +0x454
go.mondoo.com/cnspec/v9/policy/scan.(*localAssetScanner).run(0x14000d35d38)
        /Users/chris/go/src/go.mondoo.com/cnspec/policy/scan/local_scanner.go:781 +0x2c
go.mondoo.com/cnspec/v9/policy/scan.(*LocalScanner).runMotorizedAsset.func1(0x101c0d4e0?, 0x14000213180?)
        /Users/chris/go/src/go.mondoo.com/cnspec/policy/scan/local_scanner.go:632 +0x200
go.mondoo.com/cnspec/v9/internal/datalakes/inmemory.WithDb({0x101c0d4e0?, 0x14000213180?}, 0x10?, 0x14000f89df0)
        /Users/chris/go/src/go.mondoo.com/cnspec/internal/datalakes/inmemory/inmemory.go:52 +0x4c
go.mondoo.com/cnspec/v9/policy/scan.(*LocalScanner).runMotorizedAsset(0x14000041600?, 0x1016299a9?)
        /Users/chris/go/src/go.mondoo.com/cnspec/policy/scan/local_scanner.go:608 +0x68
go.mondoo.com/cnspec/v9/policy/scan.(*LocalScanner).RunAssetJob(0x140003a7b90, 0x140005ac200)
        /Users/chris/go/src/go.mondoo.com/cnspec/policy/scan/local_scanner.go:572 +0x37c
go.mondoo.com/cnspec/v9/policy/scan.(*LocalScanner).distributeJob.func2()
        /Users/chris/go/src/go.mondoo.com/cnspec/policy/scan/local_scanner.go:496 +0xc8
created by go.mondoo.com/cnspec/v9/policy/scan.(*LocalScanner).distributeJob in goroutine 1
        /Users/chris/go/src/go.mondoo.com/cnspec/policy/scan/local_scanner.go:476 +0x1324
```

**Expected behavior**

Reporting works as expected and checks are scored in both policies

**Desktop (please complete the following information):**

cnspec 9.7.1

--- a/policy/resolver.go
+++ b/policy/resolver.go
@@ -977,6 +977,9 @@ func (cache *policyResolverCache) addCheckJob(ctx context.Context, check *explor
 		cache.global.reportingJobsByUUID[uuid] = queryJob
 		cache.global.reportingJobsByMsum[check.Checksum] = append(cache.global.reportingJobsByMsum[check.Checksum], queryJob)
 		cache.childJobsByMrn[check.Mrn] = append(cache.childJobsByMrn[check.Mrn], queryJob)
+	}
+
+	if ownerJob.ChildJobs[queryJob.Uuid] == nil {
 		ownerJob.ChildJobs[queryJob.Uuid] = impact
 	}
 

--- a/policy/scan/local_scanner_test.go
+++ b/policy/scan/local_scanner_test.go
@@ -149,6 +149,49 @@ func (s *LocalScannerSuite) BeforeTest(suiteName, testName string) {
 	}
 }
 
+func (s *LocalScannerSuite) TestRunIncognito_SharedQuery() {
+	bundle, err := policy.BundleFromPaths("./testdata/shared-query.mql.yaml")
+	s.Require().NoError(err)
+
+	_, err = bundle.CompileExt(context.Background(), policy.BundleCompileConf{
+		Schema:        s.schema,
+		RemoveFailing: true,
+	})
+	s.Require().NoError(err)
+
+	s.job.Bundle = bundle
+	bundleMap := bundle.ToMap()
+
+	ctx := context.Background()
+	scanner := NewLocalScanner()
+	res, err := scanner.RunIncognito(ctx, s.job)
+	s.Require().NoError(err)
+	s.Require().NotNil(res)
+
+	full := res.GetFull()
+	s.Require().NotNil(full)
+
+	s.Equal(1, len(full.Reports))
+
+	for k, r := range full.Reports {
+		// Verify the score is 100
+		s.Equal(uint32(100), r.GetScore().Value)
+
+		p := full.ResolvedPolicies[k]
+
+		// Get the code id for all the executed queries
+		executedQueries := []string{}
+		for qCodeId := range p.ExecutionJob.Queries {
+			executedQueries = append(executedQueries, qCodeId)
+		}
+
+		expectedQueries := []string{
+			bundleMap.Queries["//local.cnspec.io/run/local-execution/queries/sshd-01"].CodeId,
+		}
+		s.ElementsMatch(expectedQueries, executedQueries)
+	}
+}
+
 func (s *LocalScannerSuite) TestRunIncognito_ExceptionGroups() {
 	bundle, err := policy.BundleFromPaths("./testdata/exception-groups.mql.yaml")
 	s.Require().NoError(err)

--- a/policy/scan/testdata/shared-query.mql.yaml
+++ b/policy/scan/testdata/shared-query.mql.yaml
@@ -1,0 +1,20 @@
+policies:
+  - uid: example1
+    name: Example policy 1
+    version: "1.0.0"
+    groups:
+      - filters: true == true
+        checks:
+          - uid: sshd-01
+  - uid: example2
+    name: Example policy 2
+    version: "1.0.0"
+    groups:
+      - filters: true == true
+        checks:
+          - uid: sshd-01
+
+queries:
+  - uid: sshd-01
+    title: Ensure the port is set to 22
+    mql: true == true

